### DISCUSSION TITLE
Real-time clock (RTC) support added to HW5

### DISCUSSION THREAD

**Discussion by Ancyker** [sanni/cartreader#827] [General Chat]
I will write the wiki how-to tomorrow. But a quick couple of things...

I've already ordered it:
![oscr-hw5-pcb-kit-1-rev5](https://github.com/sanni/cartreader/assets/7105333/3e8a87be-ed32-43ef-9b35-93f8aecd1ca7)

...and it's been tested...
![Honeycam 2023-07-15 23-43-45](https://github.com/sanni/cartreader/assets/7105333/583f93c3-f774-43ba-bfeb-0307636d9c15)

![image](https://github.com/sanni/cartreader/assets/7105333/4bfd11f7-04cb-4ece-9708-93cca9550d6b)

<img width="499" alt="Screenshot_1221" src="https://github.com/sanni/cartreader/assets/7105333/bd2730bb-8330-42b6-8927-a55923abc727">

...but there is an issue...
![2023-07-16 00 03 18](https://github.com/sanni/cartreader/assets/7105333/c08867b5-19e9-45d2-a5d3-561f0583429f)

A revision will be needed to the frame to account for the RTC battery holder. If you are not installing RTC support then this does not affect you (you can just use the existing frame). I just got a 3D printer but haven't got it all set up yet. Should be up and running within the next few weeks. If someone else doesn't fix the model by then I will do it. My printer is a resin printer as well, so I'll be making a resin-compatible frame (you can't use the heat-set inserts with resin).

Firmware support for this IC was already present, I just added a few ways to interact with it over serial so the upcoming firmware updater could set the time properly. If you are on Windows you need to set the time to your local time, if you are on Linux you'll need to set it to UTC.

I chose a CR1616 battery, but fret not. I have the RTC IC pulling power directly off of VBUS. This means if the cart reader is plugged in but turned off, the RTC will still get power off of the USB port rather than the CR1616 battery. That should prolong the life of the battery if, like me, you leave your reader plugged in all the time but turn it off when not using it.

Parts:
* RTC IC: https://www.digikey.com/en/products/detail/analog-devices-inc-maxim-integrated/DS3231MZ/2754396
* Holder: https://www.digikey.com/en/products/detail/keystone-electronics/3012/1627783
* The capacitor is the same 0.1uF cap (0402) used by vselect.

**Comment by Ancyker** [comment]
RTC guide/wiki page is up -> https://github.com/sanni/cartreader/wiki/Real%E2%80%90Time-Clock

**Comment by ccfman2004** [comment]
I see there are a few different versions of the DS3231MZ chip.  Does it matter which one I get as long as it's the 8 pin version? The DS3231MZ+TRL is the cheapest at about $2 per chip and I want to verify it will work or do I have to get the more expensive DS3231MZ/V+ or DS3231MZ/V+T.

**Reply by Ancyker** [reply]
Well, TRL is just a packaging method (Tape/Reel), as is T (Tube). They will likely all work.

**Reply by ccfman2004** [reply]
Thanks

**Reply by Ancyker** [reply]
"/V" is discontinued/"not recommended for new designs" (because people that actually need it will be stocking up), so that could be why you found the other one so cheap.

From the datasheet:
> +Denotes a lead(Pb)-free/RoHS-compliant package.
> /V denotes an automotive qualified part.

So they are all the same for our purposes.

**Reply by Ancyker** [reply]
Also, I did find a $2-3 listing as well and bought some. I do have them on hand and they appear the same, but I have not tested their quality yet which is why I did not put them on the wiki. I don't like blindly recommending things. :3
